src/phoenix/trace/llama_index/callback.py

```diff
@@ -381,3 +411,43 @@ def _get_response_output(response: Any) -> Iterator[Tuple[str, Any]]:
     else:
         yield OUTPUT_VALUE, str(response)
         yield OUTPUT_MIME_TYPE, MimeType.TEXT
+
+
+def _get_end_time(event_data: CBEventData, span_events: List[SpanEvent]) -> Optional[datetime]:
+    """
+    A best-effort attempt to get the end time of an event.
+
+    LlamaIndex's callback system does not guarantee that the on_event_end hook is always called, for
+    example, when an error occurs mid-event.
+    """
+    if end_event := event_data.get("end_event"):
+        tz_naive_end_time = _timestamp_to_tz_naive_datetime(end_event.time)
+    elif span_events:
+        last_span_event = span_events[-1]
```

Comment:
`-1` is not correct here (note that you have to reverse it on line 339). We should just call `sorted` here to ensure correctness.

Reply:
addressed in [4dd7df3](https://github.com/Arize-ai/phoenix/pull/1449/commits/4dd7df31606bcffbc8c7c42456646624351b44c2)